source/extensions/tracers/xray/tracer.cc

```diff
@@ -0,0 +1,124 @@
+#include "extensions/tracers/xray/tracer.h"
+
+#include <chrono>
+#include <ctime>
+#include <random>
+
+#include "common/common/utility.h"
+#include "common/tracing/http_tracer_impl.h"
+
+#include "extensions/tracers/xray/util.h"
+#include "extensions/tracers/xray/xray_core_constants.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+const std::string Tracer::VERSION_ = "1";
+const std::string Tracer::DELIMITER_ = "-";
+
+std::string Tracer::generateRandom96BitString() {
+  char values[25] = {'\0'};
+  auto seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+  std::mt19937 mt_rand(seed);
+
+  for (int i = 0; i < 24; i++) {
+    values[i] = hex_digits[mt_rand() % 16];
+  }
+  return values;
+}
+
+std::string Tracer::generateTraceId() {
+  time_t epoch = time(nullptr);
+  std::stringstream stream;
+  stream << std::hex << epoch;
+  std::string result(stream.str());
+  std::stringstream trace_id;
+  trace_id << VERSION_ << DELIMITER_ << result << DELIMITER_ << generateRandom96BitString();
+  return trace_id.str();
+}
```

Comment:
You replace most of this function with:
```
 return absl::StrCat(VERSION_, DELIMETER_, Hex::unit64ToHex(epoch), DELIMITER_,  generateRandom96BitString());
```
and avoid

Reply:
Thank you for your suggestion for `absl::StrCat`. However, I cannot use `Hex::unit64ToHex` since it generated 18 bits of hex string and what I need is 8 bits hex string.